torchmetrics/utilities/checks.py

```diff
@@ -604,3 +605,117 @@ def _check_retrieval_target_and_prediction_types(
+    else:
+        raise ValueError("Unknown format for comparison")
+
+
+def check_forward_no_full_state(
+    metric_class: object,
+    init_args: Dict[str, Any] = {},
+    input_args: Dict[str, Any] = {},
+    num_update_to_compare: Sequence[int] = [10, 100, 1000],
+    reps: int = 5,
+) -> bool:
+    """Utility function for checking if the new ``full_state_update`` property can safely be set to ``False`` which
+    will for most metrics results in a speedup when using ``forward``.
+
+    Args:
+        metric_class: metric class object that should be checked
+        init_args: dict containing arguments for initializing the metric class
+        input_args: dict containing arguments to pass to ``forward``
+        num_update_to_compare: if we successfully detech that the flag is safe to set to ``False``
+            we will run some speedup test. This arg should be a list of integers for how many
+            steps to compare over.
+        reps: number of repetitions of speedup test
+
+    Example (states in ``update`` are independent, save to set ``full_state_update=False``)
+        >>> from torchmetrics import ConfusionMatrix
+        >>> check_forward_no_full_state(
+        ...     ConfusionMatrix,
+        ...     init_args = {'num_classes': 3},
+        ...     input_args = {'preds': torch.randint(3, (10,)), 'target': torch.randint(3, (10,))},
+        ... ) # doctest: +SKIP
+        Full state for 10 steps took: 0.0072667645290493965+-3.828236731351353e-05
+        Partial state for 10 steps took: 0.004743088968098164+-0.0005820328951813281
+        Full state for 100 steps took: 0.0730440765619278+-0.0003615743189584464
+        Partial state for 100 steps took: 0.04705753177404404+-0.002143740188330412
+        Full state for 1000 steps took: 0.8512250781059265+-0.052338723093271255
+        Partial state for 1000 steps took: 0.5545409917831421+-0.04722180590033531
+        True
+
+    Example (states in ``update`` are dependend meaning that ``full_state_update=True``):
+        >>> from torchmetrics import ConfusionMatrix
+        >>> class MyMetric(ConfusionMatrix):
+        ...     def update(self, preds, target):
+        ...         super().update(preds, target)
+        ...         # by construction make future states dependent on prior states
+        ...         if self.confmat.sum() > 20:
+        ...             self.reset()
+        >>> check_forward_no_full_state(
+        ...     MyMetric,
+        ...     init_args = {'num_classes': 3},
+        ...     input_args = {'preds': torch.randint(3, (10,)), 'target': torch.randint(3, (10,))},
+        ... )
+        False
+    """
+
+    class FullState(metric_class):
+        full_state_update = True
+
+    class PartState(metric_class):
+        full_state_update = False
+
+    fullstate = FullState(**init_args)
+    partstate = PartState(**init_args)
+
+    equal = True
+    for _ in range(10):
```

Comment:
```suggestion
    for _ in range(num_update_to_compare[0]):
```
why do you have a fixed number here?

Reply:
does not really matter, just a number larger than 1 should be sufficient :]